### PR TITLE
Feature/hp 46 add static data models to juniper channel manager

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added static data models to juniper channel manager

### CODE REVIEW

**Review by DFSko — APPROVED** [review]

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeLimitTypes.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace HappyTravel.Komoro.JuniperChannelManager.Models.Enums;
+
+/// <summary>
+/// Age limit types
+/// </summary>
+public enum AgeLimitTypes
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeLimitTypes.cs:6]: Comment don't add much sense. When we should use 1 or 2? what is the difference

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeRangeCodes.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace HappyTravel.Komoro.JuniperChannelManager.Models.Enums;
+
+/// <summary>
+/// Age range codes
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeRangeCodes.cs:4]: Same here, what are that ranges? is it different range or different children's?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/FeatureTypes.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+/// <summary>
+/// Feature types
+/// </summary>
+public enum FeatureTypes
+{
+    Hotel = 1,
+    Room = 2,
+    Restaurant = 3,
+    AdditionalInfo = 4
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Enums/FeatureTypes.cs:11]: What should be in additional info?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    /// • Guest
+    /// </summary>
+    [XmlAttribute(AttributeName = "type")]
+    public AgeLimitTypes Type { get; init; }
+
+    /// <summary>
+    /// From age
+    /// </summary>
+    [XmlAttribute(AttributeName = "from")]
+    public int FromAge { get; init; }
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs:23]: is this years? than better add unit meashurment

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    /// <summary>
+    /// Maximum age
+    /// </summary>
+    [XmlAttribute(AttributeName = "maxAge")]
+    public int MaximumAge { get; init; }
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs:29]: Same here

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+public record AgeRange
+{
+    /// <summary>
+    /// Age range code. Possible values:
+    /// ChildA: First age range
+    /// ChildB: Second age range
+    /// ChildC: Third age range
+    /// </summary>
+    [XmlAttribute(AttributeName = "code")]
+    public AgeRangeCodes Code { get; init; }
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:18]: Why codes if it one?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    /// <summary>
+    /// Initial value of the age range.
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:21]: What does it mean initial and final?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    /// </summary>
+    [XmlAttribute(AttributeName = "fromAge")]
+    public int FromAge { get; init; }
+
+    /// <summary>
+    /// Final value of the age range.
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:27]: What is the final value? example of usage maybe

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/BoardType.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Xml.Serialization;
+
+namespace HappyTravel.Komoro.JuniperChannelManager.Models.Inventories;
+
+/// <summary>
+/// Board type
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/BoardType.cs:6]: What is board? Example in comments pls

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Category.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Xml.Serialization;
+
+namespace HappyTravel.Komoro.JuniperChannelManager.Models.Inventories;
+
+/// <summary>
+/// Hotel category
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Category.cs:6]: Same here

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/DirectPaymentCardType.cs [hunk 11]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Xml.Serialization;
+
+namespace HappyTravel.Komoro.JuniperChannelManager.Models.Inventories;
+
+/// <summary>
+/// Direct payment card type
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/DirectPaymentCardType.cs:6]: What card types could it be?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+public record Hotel
+{
+    /// <summary>
+    /// Hotel code
+    /// </summary>
+    public string HotelCode { get; init; } = string.Empty;
+
+    /// <summary>
+    /// Hotel code encrypted
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:13]: What is that for?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs [hunk 13]
```diff
@@ -0,0 +1,91 @@
+﻿using System.Net.Sockets;
+
+namespace HappyTravel.Komoro.JuniperChannelManager.Models.Inventories;
+
+public record Hotel
+{
+    /// <summary>
+    /// Hotel code
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:8]: In what system?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    /// </summary>
+    public string Name { get; init; } = string.Empty;
+
+    /// <summary>
+    /// Hotel category
+    /// </summary>
+    public Category Category { get; init; } = new();
+
+    /// <summary>
+    /// Hotel minimum age
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:38]: Age of hotel??

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    
+    /// <summary>
+    /// Images
+    /// </summary>
+    public List<Image>? Images { get; init; } = new(0);
+    
+    /// <summary>
+    /// Hotel time zone
+    /// </summary>
+    public string TimeZone { get; init; } = string.Empty;
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:90]: TimeZone string really?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Image.cs [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+
+/// <summary>
+/// Hotel image
+/// </summary>
+public record Image
+{
+    /// <summary>
+    /// Path to image file
+    /// </summary>
+    public string FileName { get; init; } = string.Empty;
```
Comment (suhan3z) [inline HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Image.cs:11]: Is it file name or url?

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeLimitTypes.cs (hunk 1 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeLimitTypes.cs:6]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeRangeCodes.cs (hunk 2 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Enums/AgeRangeCodes.cs:4]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Enums/FeatureTypes.cs (hunk 3 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Enums/FeatureTypes.cs:11]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs (hunk 4 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs:23]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs (hunk 5 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeLimit.cs:29]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs (hunk 6 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:18]: Enum types are always plural

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs (hunk 7 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:21]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs (hunk 8 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/AgeRange.cs:27]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/BoardType.cs (hunk 9 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/BoardType.cs:6]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/DirectPaymentCardType.cs (hunk 11 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/DirectPaymentCardType.cs:6]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs (hunk 12 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:13]: Don't know. We may not need this field at all. But the supplier included it in the data model.

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Category.cs (hunk 10 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Category.cs:6]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs (hunk 13 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:8]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs (hunk 14 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:38]: Fixed

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs (hunk 15 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Hotel.cs:90]: Yes, that's the string. For example: (UTC+05:00) Ekaterinburg

File: HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Image.cs (hunk 16 above)
Comment (AlexeyShok) [reply HappyTravel.Komoro.JuniperChannelManager/Models/Inventories/Image.cs:11]: The example specifies a relative path that includes the folder and file name, but without the site name.

**Pull request merged**